chore(examples): add Material UI example

KiwiKilian opened the pull request:
Fixes #2867 

As previously discussed @schiller-manuel.

schiller-manuel commented on examples/react/start-material-ui/app/api.ts:
we don't need API routes in this example, right?

SeanCassiere replied:
Yea, it makes sense to nuke it.

SeanCassiere commented on examples/react/start-material-ui/app/components/Header.tsx:
```diff
@@ -0,0 +1,21 @@
+import { AppBar, Box, Toolbar, css, styled } from '@mui/material'
+import RouterLink from '~/components/RouterLink'
+
+const StyledRouterLink = styled(RouterLink)(
```
Please document this usage of `StyledRouterLink` in https://github.com/TanStack/router/blob/main/docs/router/framework/react/guide/custom-link.md

schiller-manuel replied:
and then link to this example

KiwiKilian replied:
Did so! I'm not sure if I set the link properly and I don't know how to run a preview of the docs.

KiwiKilian commented:
If also changed the Counter button to be a link and moved the state to search params. This way multiple patterns are shown on how to use `Link` and `Button`.

KiwiKilian commented on docs/router/framework/react/guide/custom-link.md:
```diff
@@ -146,26 +146,86 @@ export const CustomLink: LinkComponent<typeof ChakraLinkComponent> = (
+import type { ButtonProps } from '@mui/material'
+import type { LinkComponent } from '@tanstack/react-router'
+
+interface MUIButtonLinkProps extends Omit<ButtonProps<'a'>, 'href'> {
```
Not sure why `href` has to be omitted? It seems to get overwritten by `createLink` anyways?

reihwald replied:
I also wondered why the example is written like this. We don't do something like than in our codebase and never faced any issues.

KiwiKilian replied:
Removed
